### ISSUE TITLE
PHAR dependencies are not scoped yet

### ISSUE THREAD

**Issue by alfredbez** [ergebnis/phpunit-slow-test-detector#334]
#### Steps required to reproduce the problem

I prepared a small repo with GitHub Actions to reproduce the issue [here](https://github.com/alfredbez/phpunit-slow-test-detector-issue/actions/runs/6416137211/job/17419286218).

[Problematic code](https://github.com/alfredbez/phpunit-slow-test-detector-issue/blob/main/tests/SomeTest.php#L12):
```php
        $parser = (new ParserFactory)->create(ParserFactory::PREFER_PHP7);
        $stmts = $parser->parse('<?php var_dump("test");');
        foreach ((new NodeFinder())->findInstanceOf($stmts, FuncCall::class) as $funcCalls) {
            $funcCalls->name->getParts();
        }
```

![image](https://github.com/ergebnis/phpunit-slow-test-detector/assets/1001186/1f0427cc-72b7-4fea-8f00-88734abce549)

The problem in this example is that `PhpParser\Node\Name` is not used from the project autoloader but from the one in the PHAR.

#### Expected Result

I would expect that dependencies of that PHAR are prefixed and do not collide with project dependencies.

#### Actual Result

Dependencies of that PHAR are not prefixed and collide with project dependencies.

**Comment by alfredbez** [comment]
I tried to add `KevinGH\\Box\\Compactor\\PhpScoper` to the [list of compactors](https://github.com/ergebnis/phpunit-slow-test-detector/blob/b20d615a57343ef1cffa6280d88a5150b65d8e20/box.json#L3-L6) as described [here](https://github.com/box-project/box/blob/main/doc/code-isolation.md#isolating-the-phar), but ran into this issue:

<details>
  <summary>Show issue</summary>

```
➜ .phive/box compile --config=box.json --debug
[debug] Changed the memory limit from "128MB" to "512M"
[debug] Checking BOX_ALLOW_XDEBUG
[debug] phar.readonly is enabled
[debug] The Xdebug extension is loaded (3.2.0) xdebug.mode=develop
[debug] Configured `phar.readonly=0`
[debug] Process restarting (BOX_ALLOW_XDEBUG=internal|3.2.0|1|*|*)
[debug] Running /opt/homebrew/Cellar/php/8.2.11/bin/php .phive/box compile --config=box.json --debug
[debug] Current memory limit: "512MB"
[debug] Checking BOX_ALLOW_XDEBUG
[debug] Restarted (113 ms). The Xdebug extension is not loaded

    ____
   / __ )____  _  __
  / __  / __ \| |/_/
 / /_/ / /_/ />  <
/_____/\____/_/|_|


Box version 4.3.8@5534406 2023-03-17 08:56:09 UTC

 // Loading the configuration file "box.json".

🔨  Building the PHAR "/Users/bez/projects/privat/phpunit-slow-test-detector/.build/phar/phpunit-slow-test-detector.phar"

[debug] Increased the maximum number of open file descriptors from ("256", "unlimited") to ("2176", "unlimited")
? Checking Composer compatibility
    > '/opt/homebrew/bin/composer' '--version'
    > 2.6.4 (Box requires ^2.2.0)
    > Supported version detected
? Setting replacement values
  + @git@: 2.4.0
? Registering compactors
  + KevinGH\Box\Compactor\Json
  + KevinGH\Box\Compactor\Php
  + KevinGH\Box\Compactor\PhpScoper
? Adding main file: /Users/bez/projects/privat/phpunit-slow-test-detector/phar/phpunit-slow-test-detector.php
? Adding requirements checker
? Adding binary files
    > No file found
? Auto-discover files? Yes
? Exclude dev files? Yes
? Adding files
[debug] Restored the maximum number of open file descriptors

In functions.php line 21:

  [_HumbugBox89e0edca2b33\Amp\Serialization\SerializationException]
  Unsupported callable: Serialization of 'ReflectionParameter' is not allowed


Exception trace:
  at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/parallel-functions/src/functions.php:21
 _HumbugBox89e0edca2b33\Amp\ParallelFunctions\parallel() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/parallel-functions/src/functions.php:31
 _HumbugBox89e0edca2b33\Amp\ParallelFunctions\{closure}() at n/a:n/a
 Generator->current() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/amp/lib/Coroutine.php:33
 _HumbugBox89e0edca2b33\Amp\Coroutine->__construct() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/amp/lib/functions.php:54
 _HumbugBox89e0edca2b33\Amp\call() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/parallel-functions/src/functions.php:30
 _HumbugBox89e0edca2b33\Amp\ParallelFunctions\parallelMap() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Box.php:229
 _HumbugBox89e0edca2b33\KevinGH\Box\Box->processContents() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Box.php:173
 _HumbugBox89e0edca2b33\KevinGH\Box\Box->addFiles() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Console/Command/Compile.php:250
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile::addFilesWithErrorHandling() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Console/Command/Compile.php:244
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile::addFiles() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Console/Command/Compile.php:147
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile->createPhar() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Console/Command/Compile.php:126
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile->execute() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/fidry/console/src/Command/SymfonyCommand.php:61
 _HumbugBox89e0edca2b33\Fidry\Console\Command\SymfonyCommand->execute() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/symfony/console/Command/Command.php:168
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Command\Command->run() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/symfony/console/Application.php:681
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->doRunCommand() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/symfony/console/Application.php:226
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->doRun() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/symfony/console/Application.php:122
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->run() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/fidry/console/src/Application/ApplicationRunner.php:34
 _HumbugBox89e0edca2b33\Fidry\Console\Application\ApplicationRunner->run() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/bin/box:47
 require() at /Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar:18

In functions.php line 19:

  [Exception]
  Serialization of 'ReflectionParameter' is not allowed


Exception trace:
  at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/parallel-functions/src/functions.php:19
 serialize() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/parallel-functions/src/functions.php:19
 _HumbugBox89e0edca2b33\Amp\ParallelFunctions\parallel() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/parallel-functions/src/functions.php:31
 _HumbugBox89e0edca2b33\Amp\ParallelFunctions\{closure}() at n/a:n/a
 Generator->current() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/amp/lib/Coroutine.php:33
 _HumbugBox89e0edca2b33\Amp\Coroutine->__construct() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/amp/lib/functions.php:54
 _HumbugBox89e0edca2b33\Amp\call() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/amphp/parallel-functions/src/functions.php:30
 _HumbugBox89e0edca2b33\Amp\ParallelFunctions\parallelMap() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Box.php:229
 _HumbugBox89e0edca2b33\KevinGH\Box\Box->processContents() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Box.php:173
 _HumbugBox89e0edca2b33\KevinGH\Box\Box->addFiles() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Console/Command/Compile.php:250
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile::addFilesWithErrorHandling() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Console/Command/Compile.php:244
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile::addFiles() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Console/Command/Compile.php:147
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile->createPhar() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/src/Console/Command/Compile.php:126
 _HumbugBox89e0edca2b33\KevinGH\Box\Console\Command\Compile->execute() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/fidry/console/src/Command/SymfonyCommand.php:61
 _HumbugBox89e0edca2b33\Fidry\Console\Command\SymfonyCommand->execute() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/symfony/console/Command/Command.php:168
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Command\Command->run() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/symfony/console/Application.php:681
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->doRunCommand() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/symfony/console/Application.php:226
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->doRun() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/symfony/console/Application.php:122
 _HumbugBox89e0edca2b33\Symfony\Component\Console\Application->run() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/vendor/fidry/console/src/Application/ApplicationRunner.php:34
 _HumbugBox89e0edca2b33\Fidry\Console\Application\ApplicationRunner->run() at phar:///Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar/bin/box:47
 require() at /Users/bez/projects/privat/phpunit-slow-test-detector/.build/phive/phars/humbug/box-4.3.8.phar:18

compile [--debug] [--no-parallel] [--no-restart] [--dev] [--no-config] [--with-docker] [--composer-bin COMPOSER-BIN] [--allow-composer-check-failure] [-c|--config CONFIG] [-d|--working-dir WORKING-DIR]

[debug] Restarted process exited 1
```

</details>

**Comment by localheinz** [comment]
Thank you for reporting, @alfredbez - fixed this in #342!
